### PR TITLE
Implemented proportional armor modifier, applied armor in other damage types and consume armor when player is damaged

### PR DESCRIPTION
**Pull request by PEMapModder**

### CODE REVIEW

**Comment by shoghicp** [comment]
Nice!
I'm going to review all of the pending pull requests today, so this might make it into 1.4.1 :)

File: src/pocketmine/Player.php [hunk 1]
```diff
@@ -2685,7 +2659,23 @@ public function attack($damage, $source = EntityDamageEvent::CAUSE_MAGIC){
 		parent::attack($damage, $source);
 
 		if($source instanceof EntityDamageEvent and $source->isCancelled()){
```
Comment (PEMapModder) [inline src/pocketmine/Player.php]: Ugh I forgot to delete `$source->isCancelled()`

Comment (shoghicp) [inline src/pocketmine/Player.php]: Can you commit a patch? You can edit that file on your branch using the GitHub web interface, it'll show up here.

Comment (PEMapModder) [inline src/pocketmine/Player.php]: is that another commit?

**Comment by iJoshuaHD** [comment]
finally! now i dont have to hardcode this into my plugin :D

**Comment by shoghicp** [comment]
I've reverted this, please open a new pull request when everything is fixed :)

**Pull request merged**